src/main/java/com/example/lombokzadaniekoncowe/WeatherService.java

```diff
@@ -0,0 +1,46 @@
+package com.example.lombokzadaniekoncowe;
+
+import com.example.lombokzadaniekoncowe.exception.CityGeoDataNotAvailableException;
+import com.example.lombokzadaniekoncowe.model.GeoCoordinates;
+import com.example.lombokzadaniekoncowe.model.WeatherIndicator;
+import com.example.lombokzadaniekoncowe.model.WeatherResponseDto;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class WeatherService {
+
+    //https://openweathermap.org/current#geo
+    public static final String API_KEY =  "15a7589e0c618768809de9771cef6e79";
+    public static final RestTemplate restTemplate = new RestTemplate();
+
+    public List<WeatherIndicator> getWeatherIndicators(String city) {
+        GeoCoordinates geoData = getGeoData(city);
+        String weatherUrl = "https://api.openweathermap.org/data/2.5/weather?lat=" + geoData.getLat() + "&lon=" + geoData.getLon() + "&units=Metric&appid=" + API_KEY;
+        try {
+            WeatherResponseDto response = restTemplate.getForObject(weatherUrl, WeatherResponseDto.class);
+            Map<String, Double> main = response.getMain();
+            return main.entrySet().stream().map(entry -> new WeatherIndicator(entry.getKey(), entry.getValue())).collect(Collectors.toList());
```

Comment:
Użycie WeatherResponseDto z mapą jest jeszcze całkiem ok, ale później powinnaś to jednak przetransformować do jakiegoś obiektu, w którym byłyby pola typu temperature, humidity itp. W Twoim podejściu mam wrażenie, że klasa WeatherIndicator tak naprawdę niewiele wnosi, bo w większości przypadków już wygodniej jest pobrać jakąś wartość z mapy, niż przeszukiwać za każdym razem listę w celu znalezienia jakiejś wartości.